platform/linux-generic/include/odp_ipsec_internal.h

```diff
@@ -101,7 +115,20 @@ struct ipsec_sa_s {
 
 		union {
 			struct {
-				odp_atomic_u64_t antireplay;
+				/* AR window lock */
+				odp_spinlock_t lock;
+
+				/* Number of buckets in AR window */
+				uint16_t nb_bucket;
+
+				/* AR window size  */
+				uint32_t win_size;
```

Comment:
It would be better to move nb_bucket and win_size out of the hot section as they stay constant and the hot section may be frequently written to by multiple threads.

Reply:
ack.